lib/ansible/modules/system/timezone.py

```diff
@@ -322,22 +322,22 @@ def __init__(self, module):
         # Validate given timezone
         if 'name' in self.value:
             tzfile = self._verify_timezone()
-            self.update_timezone  = self.module.get_bin_path('cp', required=True)
-            self.update_timezone += ' %s /etc/localtime' % tzfile
+            self.update_timezone  = ['%s %s /etc/localtime' % (self.module.get_bin_path('cp', required=True), tzfile)]
         self.update_hwclock = self.module.get_bin_path('hwclock', required=True)
         # Distribution-specific configurations
         if self.module.get_bin_path('dpkg-reconfigure') is not None:
             # Debian/Ubuntu
-            self.update_timezone       = self.module.get_bin_path('dpkg-reconfigure', required=True)
-            self.update_timezone      += ' --frontend noninteractive tzdata'
+            # With additional hack for https://bugs.launchpad.net/ubuntu/+source/tzdata/+bug/1554806
+            self.update_timezone       = ['rm -f /etc/localtime', '%s --frontend noninteractive tzdata' %
+                                          self.module.get_bin_path('dpkg-reconfigure', required=True)]
```

Comment:
If this is the temporary fix, instead of making lists, just inserting the command minimizes the diff:

```diff
             self.update_timezone      += ' --frontend noninteractive tzdata'
+            self.update_timezone       = 'rm -f /etc/localtime && ' + self.update_timezone
             self.conf_files['name']    = '/etc/timezone'
```